File: khatkit/src/main/assets/libs/std/1.0.0/main.lua
-- @khatkit/std 内置库（Lua 子集）
local M = {}

function M.join_path(...)
  local parts = { ... }
  local joined = table.concat(parts, "/")
  return (joined:gsub("//+", "/"))
end

function M.file_size(bytes)
  local units = { "B", "KB", "MB", "GB", "TB" }
  local n = tonumber(bytes) or 0
  local index = 1
  while n >= 1024 and index < #units do
    n = n / 1024
    index = index + 1
  end
  return string.format("%.1f%s", n, units[index])
end

return M

File: khatkit/src/main/assets/libs/std/1.0.0/main.js
// @khatkit/std 内置库（JS 子集）
function joinPath() {
  var parts = Array.prototype.slice.call(arguments);
  return parts.join('/').replace(/\/+/g, '/');
}

function formatSize(bytes) {
  var units = ['B', 'KB', 'MB', 'GB', 'TB'];
  var n = Number(bytes) || 0;
  var i = 0;
  while (n >= 1024 && i < units.length - 1) {
    n /= 1024;
    i++;
  }
  return n.toFixed(1) + units[i];
}

module.exports = {
  joinPath: joinPath,
  formatSize: formatSize,
};
